### DISCUSSION TITLE
Uhhhh how do I create a markdown_extensions extension?

### DISCUSSION THREAD

**Discussion by karl-police** [mkdocs/mkdocs#3491] [Q&A] [answered]
I've like tried, to create docs/extensions/myExtension.py

and include it very dumb with
```
markdown_extensions:
```
thing is I don't even know what to put in there in the first place

there's too many docs that don't really look like what I want

I don't know what to do and how to load it 🤷

**Comment by pawamoy** [comment]
Python-Markdown extensions documentation: https://python-markdown.github.io/extensions/api/.
Try to have a look at how other extensions are written :slightly_smiling_face:

**Reply by karl-police** [reply]
it's unclear to me where I have to put the file and how ``markdown_extensions`` will know about it though

**Comment by karl-police** [comment, marked as answer]
I got the extension to load now

  - docs.extensions.func_args

folder path ``docs/extensions/func_args``


```
def makeExtension(*args, **kwargs):
    # Return extension
    
    return MyExtension(*args, **kwargs)
```

there was an issue with this
this is the important thing, if this is wrong it won't work, but it will load it like that

**Reply by karl-police** [reply]
The actual issue was that I had to re-run ``mkdocs serve``

@pawamoy how can I prevent from having to do that everytime?

**Comment by oprypin** [comment]
You can make a Markdown extension isolated to one site without even registering it
 See here for an example - just add a hook and register it
https://github.com/pypa/hatch/pull/966/files#diff-96e30b390974784d0fdb81cf313f7f276328406652e2709d76a5828ed7cafb44
https://github.com/pypa/hatch/pull/966/files#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2

**Reply by karl-police** [reply]
what's the best way to parse through

```
<args>
    <arg name="arg1" type="string">Description</arg>
    <arg name="arg2" type="string">Description</arg>
</args>
```

and convert it to something else

because preprocessor has it all split up in lines and idk

**Reply by oprypin** [reply]
You should probably start over with another question and describe what you actually want to achieve, which may or may not even be best done with a preprocessor

**Reply by karl-police** [reply]
oops, my copy paste messed up while I was testing
@oprypin 


```
<args>
    <arg name="arg1" type="string">Description</arg>
    <arg name="arg2" type="string">Description</arg>
</args>
```

that's what I am trying to parse and add as an extension
